pilot/cmd/pilot-agent/status/server.go

```diff
@@ -481,24 +487,38 @@ type PrometheusScrapeConfiguration struct {
 // but we still want Envoy metrics. Instead, errors are tracked in the failed scrape metrics/logs.
 func (s *Server) handleStats(w http.ResponseWriter, r *http.Request) {
 	metrics.ScrapeTotals.Increment()
-	var envoy, application, agent []byte
 	var err error
+	var envoy, application io.ReadCloser
+	var envoyCancel, appCancel context.CancelFunc
+	defer func() {
+		if envoy != nil {
+			envoy.Close()
+		}
+		if application != nil {
+			application.Close()
+		}
+		if envoyCancel != nil {
+			envoyCancel()
+		}
+		if appCancel != nil {
+			appCancel()
+		}
+	}()
 
 	// Gather all the metrics we will merge
 	if !s.config.NoEnvoy {
-		if envoy, _, err = s.scrape(fmt.Sprintf("http://localhost:%d/stats/prometheus", s.envoyStatsPort), r.Header); err != nil {
+		if envoy, envoyCancel, _, err = s.scrape(fmt.Sprintf("http://localhost:%d/stats/prometheus", s.envoyStatsPort), r.Header); err != nil {
 			log.Errorf("failed scraping envoy metrics: %v", err)
 			metrics.EnvoyScrapeErrors.Increment()
 		}
-		// Process envoy's metrics to make them compatible with FmtOpenMetrics
```

Comment:
nit: we lost this commentary. Can we explain *why* we are doing processing. Even better if there is a link to some the  prom/openmetrics spec describing this, etc. Understand if that isn't possible though

Reply:
Blank lines are not mentioned on  [OpenMetrics spec](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md). So they're not permitted .  See https://github.com/OpenObservability/OpenMetrics/issues/250. 
[Prom spec](https://github.com/prometheus/docs/blob/512f28293efc7dba62c13dc6e9ea5247c39f27d4/content/docs/instrumenting/exposition_formats.md?plain=1#L45) are mentioned that they are be ignored. 

Actually I working on removing empty line on envoy through https://github.com/envoyproxy/envoy/pull/22403 .  But I think we should replace the `\n\n` due to compatible with old envoy version.

Reply:
istio-agent is tightly coupled with envoy versions (bundled in same.docker image) so I would actually be fine with just fixing in envoy and making this code simple